examples/passport-login/README.md

```diff
@@ -0,0 +1,112 @@
+- Link third party accounts with a LoopBack user (for example, a LoopBack user
+  can have associated facebook/google accounts to retrieve pictures).
+
+## Prerequisites
+
+Before starting this tutorial, make sure you have Client-ids/Secrets from third
+party apps
+
+- [facebook](https://developers.facebook.com/apps)
+- [google](https://console.developers.google.com/project)
+- [twitter](https://apps.twitter.com/) **Not yet implemented**
+
+## Install the example locally
+
+1. Run the `lb4 example` command to install `example-passport-login` repository:
+
+   ```sh
+   lb4 example passport-login
+   ```
+
+2. change into directory and then install the required dependencies:
+
+   ```sh
+   cd loopback4-example-passport-login && npm i
+   ```
+
+## Try it out with FaceBook
+
+### Create a test app and test user in FaceBook
+
+1. Login to facebook developer console: https://developers.facebook.com/apps
+2. Click on `My Apps` tab in the dashboard menu, and then `Add a new App`
+3. Pick the platform as `Website` and then enter app category, app name and
+   "Site URL" (Skip the quick start)
+4. Click `Settings` tab from navigation menu, note the "App ID" and "App Secret"
+   and save
+5. Click the `Roles` tab from navigation menu, then the `Test users` link under
+   it, to display a list of test users. You can also create a new test user.
+6. Click on the edit->`Change permissions granted by this test user` and add
+   [email, manage_pages] permissions
+
+- NOTE:
+  - Your app may not work if the settings are missing a contact email and/or
+    "Site URL".
+  - if you are testing locally, you can simply use `localhost:[port#]` as your
+    "Site URL".
+
+### 3. Create oauth2-providers.json
+
+- Copy `oauth2-providers.template.json` from this example project's root to
+  `oauth2-providers.json`
+- Update facebook oauth2 config with the values for `clientID/clientSecret` from
+  your test app.
+
+  ```
+  "facebook-login": {
+    "provider": "facebook",
+    "module": "passport-facebook",
+    "clientID": "xxxxxxxxxxxxxxx",
+    "clientSecret": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+    "callbackURL": "/auth/facebook/callback",
+    "authPath": "/auth/facebook",
+    "callbackPath": "/auth/facebook/callback",
+    "successRedirect": "/auth/account",
+    "failureRedirect": "/login",
+    "scope": ["email"],
+    "failureFlash": true,
+    "profileFields": ["gender", "link", "locale", "name", "timezone", "verified", "email", "updated_time"]
+  }
+  ```
+
+The `profileFields` field above tells facebook details to return in profile data
+after authentication. For more information regarding the providers template, see
+http://loopback.io/doc/en/lb2/Configuring-providers.json.html.
+
+### 6. Run the application
+
+By default the user data is stored using a memory connector and saved locally to
+`data/db.json`
+
+Start the application with
+
+```
+$ npm start
+```
+
+- Open your browser to the example app with, `http://localhost:3000`
```

Comment:
@deepakrkris , thanks for making the UI changes and addressing my earlier comments.

Let's update this section with the latest UI changes in mind (some of the instructions are out of date.)

But also it is important to register a local user BEFORE any social media login, otherwise there will be issues.

1. Open browser to localhost:3000
2. Sign up as a local user with the email of a facebook test user you created above, and provide a password for local purposes. This registers a local user, and it will eventually be linked to social media credentials in the following steps.
3. Login as this local user. You will be brought to the "View Account" tab ( and describe what they should see)
4. Click on Sign Up or Login, and the pages indicate you've already logged in.
5. Click on [Current User] button, and it calls https://localhost:3000/api/whoAmI endpoint and displays currently authenticated user.
6. Click on Logout.
7. Click on View Account, and you are not permitted since you are logged out.
8. Look at the db.json and mention what they should see in there. User with specific email, and in UserCredentials, the password you enterered when you signed up.
9. Now click on Login, and choose `Login with Facebook`. Enter the email of your facebook test user (same email you entered above), but specify the test user's facebook password (not local password).
10. Upon successful FaceBook login, you are brought to `View Account`. (Explain the data they should see) 
11. Click on Sign Up or Login, and the pages indicate you are already logged in.
12. Click on [Current User] to view the currently authenticated user.
13. Click on Logout 
14. Click on `View Account`, and you are not allowed.
15. Look at the db.json, there is still one user. It still has the UserCredentials entry as earlier, But now there is an entry in the UserIdentity section which contains the facebook profile info.
At any time (authenticated or not), open http://localhost:3000/api/ping in the browser and you should see the ping info returned. This endpoint doesn't require authentication.

Please re-word my list above to your liking. But we need to hold the user's hand to explain our UI to have a sense of what the ui can do based on the endpoints we've secured. As well as the database info that helps tie the local and social stuff together.

Reply:
Also

![image](https://user-images.githubusercontent.com/6864736/79473762-930ab180-7fd3-11ea-9cdc-55529d221bb2.png)

Reply:
@emonddr agreed, this is fixed now

Reply:
> @deepakrkris , thanks for making the UI changes and addressing my earlier comments.
> 
> Let's update this section with the latest UI changes in mind (some of the instructions are out of date.)
> 
> But also it is important to register a local user BEFORE any social media login, otherwise there will be issues.
> 
> 1. Open browser to localhost:3000
> 2. Sign up as a local user with the email of a facebook test user you created above, and provide a password for local purposes. This registers a local user, and it will eventually be linked to social media credentials in the following steps.
> 3. Login as this local user. You will be brought to the "View Account" tab ( and describe what they should see)
> 4. Click on Sign Up or Login, and the pages indicate you've already logged in.
> 5. Click on [Current User] button, and it calls https://localhost:3000/api/whoAmI endpoint and displays currently authenticated user.
> 6. Click on Logout.
> 7. Click on View Account, and you are not permitted since you are logged out.
> 8. Look at the db.json and mention what they should see in there. User with specific email, and in UserCredentials, the password you enterered when you signed up.
> 9. Now click on Login, and choose `Login with Facebook`. Enter the email of your facebook test user (same email you entered above), but specify the test user's facebook password (not local password).
> 10. Upon successful FaceBook login, you are brought to `View Account`. (Explain the data they should see)
> 11. Click on Sign Up or Login, and the pages indicate you are already logged in.
> 12. Click on [Current User] to view the currently authenticated user.
> 13. Click on Logout
> 14. Click on `View Account`, and you are not allowed.
> 15. Look at the db.json, there is still one user. It still has the UserCredentials entry as earlier, But now there is an entry in the UserIdentity section which contains the facebook profile info.
>     At any time (authenticated or not), open http://localhost:3000/api/ping in the browser and you should see the ping info returned. This endpoint doesn't require authentication.
> 
> Please re-word my list above to your liking. But we need to hold the user's hand to explain our UI to have a sense of what the ui can do based on the endpoints we've secured. As well as the database info that helps tie the local and social stuff together.

@emonddr I have incorporated these steps as suitable for user readability. please review.